pkg/utils/time.go

```diff
@@ -0,0 +1,174 @@
+package utils
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+)
+
+type Time time.Time
+
+// UnmarshalJSON Parses the json string in the custom format
+func (ct *Time) UnmarshalJSON(b []byte) (err error) {
+	s := strings.Trim(string(b), `"`)
+	nt, err := GetValidDate(s)
+	*ct = Time(nt)
+	return
+}
+
+// MarshalJSON writes a quoted string in the custom format
+func (ct Time) MarshalJSON() ([]byte, error) {
+	return []byte(ct.String()), nil
+}
+
+// String returns the time in the custom format
+func (ct *Time) String() string {
+	t := time.Time(*ct)
+
+	return fmt.Sprintf("%q", t.Format(time.RFC3339))
+}
+
+func GetValidDate(timeString string) (time.Time, error) {
+
+	date, err := ParseDate("01/02/2006", timeString)
+	if err == nil {
+		return date, nil
+	}
+
+	date, err = ParseDate("01/02/06", timeString)
+	if err == nil {
+		return date, nil
+	}
```

Comment:
I'm wondering if adopting a pattern like this is better.  Any thoughts? 

https://github.com/perkeep/perkeep/blob/dd14e9ea063054c9230218c2311cea15ba7f9284/pkg/importer/feed/parse.go#L443